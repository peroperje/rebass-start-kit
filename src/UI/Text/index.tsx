import * as rebass from 'rebass';

export type TextProps = rebass.TextProps;

const Text = rebass.Text;

export default Text;
