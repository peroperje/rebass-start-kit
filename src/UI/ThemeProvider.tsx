import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
