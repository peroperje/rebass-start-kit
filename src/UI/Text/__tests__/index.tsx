import React from 'react';
import render from 'react-test-renderer';

import ThemeProvider from '../../ThemeProvider';

import Text from '../index';

describe('Text', () => {
    it('Default snapshot', () => {
        const tree = render
            .create(
                <ThemeProvider>
                    <Text>Test</Text>
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
