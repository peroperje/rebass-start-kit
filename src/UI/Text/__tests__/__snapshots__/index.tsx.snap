// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Default snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="emotion-0"
>
  Test
</div>
`;
